floss/stackstrings.py

```diff
@@ -57,10 +64,33 @@ def apicall(self, emu, op, pc, api, argv):
         stack_buf = emu.readMemory(stack_top, stack_size)
         self.ctxs.append(CallContext(op.va, stack_top, stack_bottom, stack_buf))
 
-
-def extract_call_contexts(vw, fva):
+    def posthook(self, emu, op, endpc):
+        self.get_context_via_mov_heuristic(emu, op, endpc)
+
+    def get_context_via_mov_heuristic(self, emu, op, endpc):
+        """ Extract contexts at end of a basic block (bb) if bb contains enough movs to stack memory. """
+        # TODO check number of written bytes?
+        # count movs, shortcut if this basic block has enough writes to trigger context extraction already
+        if self.mov_count < MIN_NUMBER_OF_MOVS and self.is_stack_mov(emu, op):
```

Comment:
i like the short-circuit here, where the `is_stack_mov()` is not called if the threshold is already hit. 

this means that the `self.mov_count` number will not be perfectly accurate all the time (might be lower than the actual count). but i think this is ok, with a bit of documentation.